bundled/tbb-2018_U2/src/tbb/cache_aligned_allocator.cpp

```diff
@@ -248,9 +248,10 @@ bool __TBB_EXPORTED_FUNC is_malloc_used_v3() {
     }
     __TBB_ASSERT( MallocHandler!=&DummyMalloc && FreeHandler!=&DummyFree, NULL );
     // Cast to void avoids type mismatch errors on some compilers (e.g. __IBMCPP__)
-    __TBB_ASSERT( !(((void*)MallocHandler==(void*)&malloc) ^ ((void*)FreeHandler==(void*)&free)),
+    __TBB_ASSERT( !(((void*)MallocHandler==(void*)static_cast<void*(*)(size_t)>(malloc)) ^
+                    ((void*)FreeHandler==(void*)static_cast<void(*)(void*)>(free))),
```

Comment:
Again, `free` is ambiguous because there exists a device-callable variant.

Reply:
Is this fixed in upstream TBB? That is, do you happen to know what happens if we use a system-provided TBB?

Reply:
Looks like this hasn't been touched since 2017 in TBB 2020, see https://github.com/oneapi-src/oneTBB/blame/eca91f16d7490a8abfdee652dadf457ec820cc37/src/tbb/cache_aligned_allocator.cpp#L238-L244. This function doesn't appear in `oneTBB 2021.1.1`, though.

Reply:
@bangerth I just checked that I can compile with an external oneTBB 2021.9.